src/Print.hs

```diff
@@ -122,3 +124,58 @@ prettyEnv e = case e of
   Timestamp -> "TIMESTAMP"
   This -> "THIS"
   Nonce -> "NONCE"
+
+-- | Invariant predicates are represented internally as a pair of timed
+-- expressions, one over the prestate and one over the poststate.  This is good
+-- since it keeps untimed expressions away from the various backends, and
+-- maintains a nice seperation between the various compilation passes, but
+-- unfortunately requires us to strip the timing out if we want to print the
+-- invariant in a way that is easily digestible by humans, requiring a less
+-- elegant implementation here than might be hoped for...
+prettyInvPred :: InvariantPred Timed -> String
+prettyInvPred = prettyExp . stripTime . fst
+  where
+    stripTimeTyped :: TypedExp t -> TypedExp Untimed
+    stripTimeTyped (ExpInt e) = ExpInt (stripTime e)
+    stripTimeTyped (ExpBool e) = ExpBool (stripTime e)
+    stripTimeTyped (ExpBytes e) = ExpBytes (stripTime e)
+
+    -- | Strip timing from an annotated expression, sometimes useful for display in the ui
+    stripTime :: Exp a t -> Exp a Untimed
+    stripTime e = case e of
```

Comment:
How about changing `stripTime` to `untime`? Or even `obscure` (as opposed to `annotate` lol sorry)?